src/unix/inc/mincom/ccomptrs.h

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// ccomptrs.h base types for managing basic reference counting and 
+// casting of IUnknown types useful for when the entire platform
+// abstraction layer, or Abstract Template Libraries are not available.
+
+#pragma once
+
+template <class T>
+class _NoAddRefReleaseOnCComPtr : 
+    public T
+{
+    private:
+    
+    ~_NoAddRefReleaseOnCComPtr() {}
+    
+    private:
+        STDMETHOD_(ULONG, AddRef)()=0;
+        STDMETHOD_(ULONG, Release)()=0;
+};
+
+
+//CComPtrBase provides the basis for all other smart pointers
+//The other smartpointers add their own constructors and operators
+template <class T>
+class CComPtrBase
+{
+protected:
+	CComPtrBase()
+	{
+		p = NULL;
+	}
+	CComPtrBase(int nNull)
+	{
+		(void)nNull;
```

Comment:
>(void)nNull; [](start = 2, length = 12)

what's the purpose of this expression?

Reply:
Probably not needed. I adapted this from the PAL. I can get rid of it.